### DISCUSSION TITLE
See where Block Patterns were used

### DISCUSSION THREAD

**Discussion by MadtownLems** [WordPress/gutenberg#49922] [Q&A]
We manage multisites of hundred of sites, and our Block Patterns are very popular with our content providers. We'd love a way to see where/how people are making use of them, but are struggling to find an elegant way to do that that doesn't disrupt the editor experience for most users.

I first thought about adding a custom class to the outermost block in the pattern, such as "block-pattern-foo", but that class shows up as an Additional CSS class in the advanced panel.

Then I tried adding it as a data-attribute to the outermost div (group) in the block pattern, but that results in invalid block markup.

I also tried adding a custom property in the markup, like:
`<!-- wp:group {"block-pattern-name":"block-pattern-foo","layout":{"type":"constrained"}} -->`

but that gets stripped upon saving.

Any suggestions for how this might be accomplished?

**Comment by t-hamano** [comment]
Hi @MadtownLems,

If you want to give a specific block an arbitrary attribute, you may use `blocks.registerBlockType` or `blocks.getSaveContent.extraProps` hooks.

Below is a file that adds anchor support to a block:

https://github.com/WordPress/gutenberg/blob/bc70b39ff920ce6416f52f10455169bf1e003307/packages/block-editor/src/hooks/anchor.js#L156-L166

However, when a pattern is inserted into the content, it is no longer a pattern but a combination of blocks. Also, once a pattern has been inserted, the user might edit the blocks inside to the point that they are no longer original.

One approach that comes to mind is to use the [parse_blocks()](https://developer.wordpress.org/reference/functions/parse_blocks/) function. It is an approach that parses all blocks of the content and considers that if one block has a configuration that exactly matches any pattern, then it is using the pattern you provided.
